src/NuGet.Core/NuGet.Configuration/Settings/SettingFactory.cs

```diff
@@ -96,13 +97,81 @@ internal static SettingBase Parse(XNode node, SettingsFile origin)
             return null;
         }
 
+        private class SettingElementComparer : IComparer<SettingElement>, IEqualityComparer<SettingElement>
```

Comment:
Should we rename to something like `SettingElementKeyComparer` as this really is only comparing the keys and we wouldn't want others to be confused about what it does.